frontend/src/components/NavStack/index.tsx

```diff
@@ -0,0 +1,57 @@
+import { IconType } from "react-icons";
+import { FaTachometerAlt, FaStream, FaUsers, FaCog } from "react-icons/fa";
+import { Link } from "react-router-dom";
+
+interface NavItemProps {
+  buttonName: string;
+  buttonDest: string;
+  IconDest: IconType;
+}
+
+const NavItem: React.FC<NavItemProps> = ({ buttonName, buttonDest, IconDest }) => {
+  return (
+    <Link to={buttonDest} className="nav-link">
+      <IconDest /> {buttonName}
+    </Link>
+  );
+};
```

Comment:
There's not much point in having this wrapper class to directly map the parameters onto the element.